### ISSUE TITLE
BUG EmailTest non-warnings are preventing tests running

### ISSUE THREAD

**Issue by tractorcow** [silverstripe/silverstripe-framework#6954]
Non-warnings (that appear as warnings) from the following method are failing due to the below exception

```php
if ($callOriginalConstructor === false && $callOriginalMethods === true) {
    throw new PHPUnit_Framework_MockObject_RuntimeException(
        'Proxying to original methods requires invoking the original constructor'
    );
}
```

However in PHPUnit results these show up as warnings "W".

```
vendor/bin/phpunit ./framework/tests/php/Control/Email/EmailTest.php 
PHPUnit 5.7.20 by Sebastian Bergmann and contributors.

....WWW..............................                             37 / 37 (100%)

Time: 2.74 seconds, Memory: 135.25MB

There were 3 warnings:

1) SilverStripe\Control\Tests\Email\EmailTest::testSendPlain
Proxying to original methods requires invoking the original constructor

/Users/dmooyman/Sites/ss40test/vendor/phpunit/phpunit/phpunit:52

2) SilverStripe\Control\Tests\Email\EmailTest::testSend
Proxying to original methods requires invoking the original constructor

/Users/dmooyman/Sites/ss40test/vendor/phpunit/phpunit/phpunit:52

3) SilverStripe\Control\Tests\Email\EmailTest::testRenderedSend
Proxying to original methods requires invoking the original constructor

/Users/dmooyman/Sites/ss40test/vendor/phpunit/phpunit/phpunit:52
```

@dhensby can you please advise? I know we've discussed these issues before, but the upgrade to phpunit 5.7 means we need to re-write these tests.

I can confirm that the above tests are exiting due to exceptions, and not due to "real" warnings that can be ignored.

**Comment by chillu** [comment]
Oi @tractorcow, please triage issues you create yourself (add labels) :D

**Comment by dhensby** [comment]
I can confirm that if you place a `$this->fail()` after the line that causes the warning the test does not fail - thus the tests are not being run.

I've tried to get the tests working with the original constructor being executed (from what I see there's no real reason this shouldn't work) but I'm getting the following error, which I'm yet to find a way to resolve.

https://github.com/sebastianbergmann/phpunit/issues/1652

**Comment by tractorcow** [comment]
I think you'll need to change the email class directly if it can't be mocked without modification.

**Comment by dhensby** [comment]
fixed  in #7490 - thanks @tractorcow
